### PR TITLE
Add WithReaderContent PathOp

### PR DESCRIPTION
**Pull request by icholy**
#160

### CODE REVIEW

File: fs/ops.go [hunk 1]
```diff
@@ -61,6 +61,22 @@ func WithBytes(raw []byte) PathOp {
...
+	return func(path Path) error {
+		if m, ok := path.(manifestFile); ok {
+			m.SetContent(ioutil.NopCloser(r))
+			return nil
+		}
+		f, err := os.OpenFile(path.Path(), os.O_WRONLY, defaultFileMode)
+		if err != nil {
+			return err
+		}
+		_, err = io.Copy(f, r)
```
Comment (dnephin) [inline fs/ops.go]: I think we need to `f.Close()` here before returning to close the file, right ?

**Comment by icholy** [comment]
We might as well check the Close error.

**Comment by dnephin** [comment]
`io.Copy` won't panic right? So calling it after the copy, but before the return should be safe.

**Comment by icholy** [comment]
@dnephin the `io.Reader` implementation could potentially panic, but that's not something I've ever actually seen.

**Comment by dnephin** [comment]
Good point! changing to defer would be good

**Pull request merged**